BizHawk.Client.EmuHawk/tools/GyroscopeBot/GyroscopeBot.cs

```diff
@@ -704,26 +708,49 @@ private void Update(bool fast)
 					if (get_round_result() == "P1")
 					{
 						_wins = _wins + 1;
-						_lastResult = "Win";
+						_lastResult = "P1 Win";
+						_p2_losses = _p2_losses + 1;
 
 					}
 					else
 					{
 						_losses = _losses + 1;
-						_lastResult = "Loss";
+						_lastResult = "P1 Loss";
+						_p2_wins = _p2_wins + 1;
 					}
 
 					_winsToLosses = (float)_wins / _totalGames;
+					_p2_winsToLosses = (float)_p2_wins / _totalGames;
 					GlobalWin.OSD.ClearGUIText();
-					GlobalWin.OSD.AddMessageForTime("Game #: " + _totalGames + " Wins: " + _wins + " Losses: " + _losses + " last result: " + _lastResult + " ratio: " + _winsToLosses, _OSDMessageTimeInSeconds);
+					GlobalWin.OSD.AddMessageForTime("Game #: " + _totalGames + " | Last Result: " + _lastResult + " | P1 Wins-Losses: " + _wins + "-" + _losses + " (" + _winsToLosses + ") | P2 Wins-Losses: " + _p2_wins + "-" + _p2_losses + " (" + _p2_winsToLosses + ")", _OSDMessageTimeInSeconds);
 				}
 
 				string command_type = "";
 				do
 				{
 					// send over the current game state
-					ControllerCommand command = SendEmulatorGameStateToController();
+					ControllerCommand command = SendEmulatorGameStateToController(this.client);
+					ControllerCommand command_p2;
+
 					command_type = command.type;
+
+					if (Global.Config.use_two_controllers)
+					{
+						do
+						{
+							command_p2 = SendEmulatorGameStateToController(this.client_p2);
+							//Console.WriteLine("p1 command: "  + command.type + "p2 command: "  + command_p2.type);
+							// send the game state to the controller
+							// if the commands don't match, wait until they do.
+						} while (command_p2.type != command_type);
+						if (command_type == "buttons")
+						{
+							command.p2 = command_p2.p1;
```

Comment:
this line is slightly confused for me. Can you put an explainer comment in here? Then LGTM!